### DISCUSSION TITLE
Dynamic Paramethers to definePageMeta

### DISCUSSION THREAD

**Discussion by vietanhbui** [nuxt/nuxt#22069] [Questions]
Hello,

I would like to add dynamic meta to definePageMeta depends on localStorage but seem not working

`<script setup>
definePageMeta({
  layout: localStorage.getItem("token") ? "default" : "guest"
});
</script>`

**Comment by gerkereitz** [comment]
Could you clarify what is not working like it should? Are you getting an error message? Is it always showing the default or guest layout when it should show the other?

I have created a quick little project where I test a dynamic layout and for me you line of code is working fine. You just have to make sure that you disable server side rendering in either the entire app or the specific page where you are accessing localStorage, because localStorage is only known on the client side and not the server side. For my example below I have disabled SSR for the entire app.

![Nuxt3 dynamic layout example](https://github.com/nuxt/nuxt/assets/58658920/9e58a810-3c55-4d4c-b34e-b8e18a938b76)

![Nuxt config SSR false](https://github.com/nuxt/nuxt/assets/58658920/131373a9-79d3-4ca2-976c-b87fb2616c0b)

![Dynamic layout test](https://github.com/nuxt/nuxt/assets/58658920/d08cbe51-2c32-4b44-a332-56d6b0625fbf)

![Dynamic layout default](https://github.com/nuxt/nuxt/assets/58658920/41c2d2b4-a026-412d-a43c-858652cd3a8c)

**Reply by vietanhbui** [reply]
Hi @gerke100 
I disabled ssr mode, and got 2 layouts similar to yours. But when I'm on the Home screen, I have a button and when I press that button it setsLocalStorage and navigates to the /test screen. then from the test screen i navigated the home screen again using router.push, now the home screen still has the guest layout, not the default layout

**Reply by vietanhbui** [reply]
only when I reload the page does the default layout work

**Reply by cannap** [reply]
can you create a repo how you have it right now please is it a global layout change or only for some pages?
when its global you can do a different way and work with localStorage event and setPagelayout 

you can also use "useState" 

then you can watch the useState where you need to change the layout and and use setPageLayout or the component NuxtLayout with name prop

**Reply by gerkereitz** [reply]
@vietanhbui  Take a look at https://nuxt.com/docs/guide/directory-structure/layouts#changing-the-layout-dynamically

By using this, the changes should apply dynamically without having to reload the page.

Live example: https://nuxt.com/docs/examples/features/layouts

**Comment by cannap** [comment]
you say depends on "router params"
but you use localstorage
https://stackblitz.com/edit/github-hw9mwk?file=pages%2Findex.vue,layouts%2Ftest.vue,layouts%2Fselect.vue 

you can add for this demo 

?layout=test
or 
?layout=select to the url 

but maybe there are other ways but we need more infos for the "use case"

**Reply by vietanhbui** [reply]
Sorry, its depends on localStorage, or state in app, i edited the title
